### DISCUSSION TITLE
Umi & Bigfish 想法征集 01

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10516] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by stormslowly** [comment]
1.  Bigfish 文档同步 通过 ACI  定期 crontab 执行，自动提 MR 和 发钉消息合并 （why ：1. 文档保鲜 2. 防止忘记，一次同步过多导致变化太大 ，review 容易出错）
2. 文档内容，增加 Live Example，尤其是第三方的 API，一站式的看了 Umi 文档就会用了，不需要再跳转源文档从头学习（why：营造一种 Umi 让我变强了个感觉 ）
3.  定个文档的规范的草案出来 (why 最近频繁做文档想到的 )
        比如 什么时候是 Umi 什么时候 是 umi ，
        代码片段的文件名用 ` ```ts title="file/path/name.ts" ` ，而不是代码片段中注释
        使用自带的高亮功能 指定需要注意的代码。

**Reply by sorrycc** [reply]
👍🏻，2、3 已加入需求文档的「文档优化」部分。

**Comment by hanzebang** [comment]
- Bigfish/Bigfish-Doctor 增加数据上报
  - Bigfish 遇到构建时/运行时错误进行上报（了解用户在使用过程中经常遇到哪些问题）
  - Bigfish-Doctor 运行数据上报（了解用户项目目前可能存在哪些问题，是否有遗漏的，是否为用户解决了问题）
  - 需要一个数据收集平台（待调研），兜底：官网应用/yuque

**Reply by sorrycc** [reply]
👍🏻，已加入需求文档。

**Comment by xiaohuoni** [comment]
- antd plugin 支持 antd@5 主题
  - 动态主题这块 antd@5 要求在 jsx 中处理，也就是这些功能需要写在 `ConfigProvider` 中，也就是现在的 `runtime` 文件。需要打通。
  - 除了包裹 `ConfigProvider` 还需要包裹 `App` 组件，用于管理一些游离的 dom 的样式，比如文字的颜色这些
- umi test 需要提供 `RootContainer` 用于组件中的单元测试，比如在组件中使用到国际化时，需要包裹国际化的 `Provider`。当前的 `TestBrowser` 不适用这个场景。

**Reply by sorrycc** [reply]
感觉可以写篇小作文，展开描述下问题和期望的功能。

**Reply by xiaohuoni** [reply]
## 关于 antd 的主题
需要写的代码见这里 https://github.com/xiaohuoni/umisse/blob/master/src/layouts/index.tsx#L34
动态主题的效果见这里 http://umisse.alitajs.com/
## 关于 test
现在直接 render 一个带有 `import { useIntl } from 'umi';` 的简单组件是不行的。TestBrowser 只能用于测试 pages 下的页面级别的组件。

**Comment by goo-yyh** [comment]
deadCode 配置在一些历史项目中会有比较大的作用，能不能单独做个 command 检测而不是嵌入在 build 里，最好再输出个 JSON 文件，可以方便那些需要批量删除的用户进行参考。

**Reply by fz6m** [reply]
借助这个工具 [umi-plugin-circular-check](https://github.com/umijs/umi-plugin-circular-check) 里的实现思路，是可以拿到项目所有引用文件的列表，扫一下 src 的文件，剩下的就是未使用的。

**Comment by wyy0512** [comment]
升级文档可以增加插件 API 变更的部分，同时为废弃的 API 提供替代方案

**Comment by txp1035** [comment]
自动化文档生成工具，主要针对api相关的，基于gihub action自动判断是不是改了对应文件自动执行脚本进行更新。哪儿些api是要输出到文档的哪儿些api是不输出到文档的，只给内部开发看的，比如configResolveMode这个api文档上没有，具体我也没有去数有多少。我觉得这块应该规范起来，就算文档查不到在代码里也需要定义好意思，这样不至于不懂的api要去看下实现才知道是做什么的。这一点插件就很好，有个description来描述是干嘛的

**Comment by sorrycc** [comment]
- （构建提速）用 [SharedArrayBuffer - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) 改进 MFSU 实现，他允许从多个不同的worker同时访问一个内存块。
- （Bug）prepare 没有覆盖通过 umi > exports > plugin-xxx/index.ts 链路引入的文件
- （Feature）需要为最佳实践调整下 umi g 的内容，1）不能生成 less 文件了，2）包含更多现有最佳实践的 g 。
- （内部）Bigfish Checker 完善，1）数量问题，2）给应用打标，比如没权限的、测试应用、phantomDependency 问题应用等，3）更好看的测试报告

**Comment by PeachScript** [comment]
1. @umijs/testing 支持生成 vitest 配置：相比于 jest，vitest 对 pure esm 的支持更好，且速度更快
2. 自动 Changelog 生成：参考 ant-design 的 [PR 模板](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)，要求提供 Changelog 信息，在发版时自动生成以提高发版效率；Umi 可直接爬 PR 内容，Bigfish 得看下怎么读 AntCode 并更新 CHANGELOG.md

**Reply by sorrycc** [reply]
1 已加入需求文档。
